gap/attributes/attr.gi

```diff
@@ -407,6 +407,9 @@ end);
 InstallMethod(StructureDescription, "for a group as semigroup",
 [IsGroupAsSemigroup],
 function(S)
+  if IsGroup(S) then
+    TryNextMethod();
+  fi;
   return StructureDescription(Range(IsomorphismPermGroup(S)));
 end);
 
```

Comment:
This chunk is relevant - without it, the tests fail.

I added several `InstallTrueMethod`s which concern `IsSemigroupWithCommutingIdempotents`. One of them means that `RankFilter(IsGroupAsSemigroup)` is now higher than `RankFilter(IsGroup)` is at the point that GAP is loaded (and the default `StructureDescription` method for `IsGroup` is installed). Thus, for any group in the category `IsGroup`, method selection now chooses this `IsGroupAsSemigroup` method for `StructureDescription`... which calls itself, ad infinitum. Many tests were failing after encountering such a recursion depth trap.

The `TryNextMethod` avoids this recursion.